index.d.ts

```diff
@@ -0,0 +1,66 @@
+/// <reference path="./typings/index.d.ts" />
+
+declare module "jsonwebtoken" {
+
+    export type Algorithm = 'RS256' | 'RS384' | 'RS512' | 'ES256' | 'ES384' | 'ES512' | 'HS256' | 'HS384' | 'HS512' | 'none';
+
+    export interface SignOptions {
+
+        algorithm?: Algorithm;
+        expiresIn?: Number | String;
+        notBefore?: Number | String;
+        audience?: String;
+        issuer?: String;
+        jwtid?: String;
+        subject?: String;
+        noTimestamp?: Boolean;
+        header?: Object;
+    }
+
+    export type SignCallback = (err: any, token: String) => void;
+
+    export interface VerifyOptions {
+
+        algorithms?: Algorithm[];
+        audience?: String;
+        issuser?: String;
+        ignoreExpiration?: Boolean;
+        subject?: String;
+        clockTolerance?: Number;
+        maxAge?: Number | String;
+        clockTimestamp?: Number;
+    }
+
+    export type VerifyCallback = (err: any, payload: Object | Buffer | String) => void;
+
+    export interface DecodeOptions {
+
+        json?: Boolean;
+        complete?: Object;
+    }
+
+    /**
+     * Generate a JSON Web Token string based on the given payload and options and proceed with the callback function
+     * @param {Object|Buffer|String} payload - Object literal, buffer or string. Please note that exp is only set if the payload is an object literal
+     * @param {Buffer|String} secretOrPrivateKey - String or buffer containing either the secret for HMAC algorithms, or the PEM encoded private key for RSA and ECDSA
```

Comment:
This could be an object as well `{ key: String | Buffer, passphrase: String }` in case the private key is password-encrypted.